jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/importer/parser/PayloadParser.java

```diff
@@ -90,4 +105,35 @@ public void writeTo(OutputStream os) throws IOException {
     os.write(payload, 0, payloadLength);
   }
 
+
+
+   public byte[] removeHeader(byte[] packet) {
+      String hexPacket = HexUtils.toHexString(packet);
+      String hexPacketParsed = "";
+      for (String key : msgStartFlags.keySet()) {
+         String startFlag = msgStartFlags.get(key);
+         int startIndex = hexPacket.toLowerCase().indexOf(startFlag);
+         if (hexPacketParsed == ""){
```

Comment:
You could use a break after a startIndex has been found that isn't -1 and the hexPacketParsed has been assigned a new value. The max number of iterations is 3 though so performance gains would be negligible. Just something to think about when writing a loop like this.